### PR TITLE
feat: Detect deprecated AMIs

### PR DESCRIPTION
**Pull request by kbeniwal**
Fixes Issue #63 

Old awsacfg:

>       reservations:
>       - instances:
>         - httpPutResponseHopLimit: 2
>           publicDnsName: ec2-11-111-11-11.us-west-1.compute.amazonaws.com

New awsacfg:

>       reservations:
>       - instances:
>         - amazonMachineImage:
>             architecture: x86_64
>             creationTime: "2023-03-05T17:40:14.000Z"
>             deprecationTime: "2025-03-05T17:40:14.000Z"
>             id: ami-0efaccd6711a132b2
>             location: amazon/amazon-eks-node-1.25-v11111111
>             name: amazon-eks-node-1.25-v11111111
>             ownerId: "111111111111"
>             platformDetails: Linux/UNIX
>             public: true
>             state: available
>             type: machine
>           httpPutResponseHopLimit: 2
>           publicDnsName: ec2-11-111-11-11.us-west-1.compute.amazonaws.com

### CODE REVIEW

File: charts/kyverno-aws-adapter/crds/security.nirmata.io_awsadapterconfigs.yaml [hunk 1]
```diff
@@ -384,4 +409,4 @@ spec:
     served: true
     storage: true
     subresources:
-      status: {}
+      status: {}
```
Comment (pns-nirmata) [inline charts/kyverno-aws-adapter/crds/security.nirmata.io_awsadapterconfigs.yaml:412]: Github suggesting a newline here.

**Review by pns-nirmata — APPROVED** [review]
Nicely done Kushal. Sometime, we should take up the code refactoring as per https://github.com/nirmata/kyverno-aws-adapter/issues/52. Sooner than later, else the main file will keep growing.

File: config/samples/policies/check-ami-deprecation-time.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  background: true
+  rules:
+  - name: check-ami-deprecation-time
+    match:
+      any:
+      - resources:
+          kinds:
+          - AWSAdapterConfig
+    validate:
+      message: "AMIs must not be used after their deprecation time"
```
Comment (anusha94) [inline config/samples/policies/check-ami-deprecation-time.yaml:23]:
```suggestion
      message: "This rule audits for AMIs that are past their deprecation time"
```

**Review by anusha94 — COMMENTED** [review]
Looks great! Just a minor change needed in the message.

**Comment by kbeniwal** [comment]
> Sometime, we should take up the code refactoring as per https://github.com/nirmata/kyverno-aws-adapter/issues/52. Sooner than later, else the main file will keep growing.

Yes, this even I felt when making changes in the Reconcile() method, as it's making the code harder to manage and navigate. Thank you for bringing this up, we should definitely address this as soon as possible.

**Pull request merged**